src/metabase/upload.clj

```diff
@@ -199,26 +199,30 @@
   [s]
   (str/replace s currency-regex ""))
 
-(defn- parse-plain-number [s]
-  (case (get-number-separators)
-    ("." ".,") (. (NumberFormat/getInstance (Locale. "en" "US")) parse s)
-    ",." (. (NumberFormat/getInstance (Locale. "de" "DE")) parse s)
-    ", " (. (NumberFormat/getInstance (Locale. "fr" "FR")) parse (str/replace s \space \u00A0)) ; \u00A0 is a non-breaking space
-    ".’" (. (NumberFormat/getInstance (Locale. "de" "CH")) parse s)))
+(let [us (NumberFormat/getInstance (Locale. "en" "US"))
+      de (NumberFormat/getInstance (Locale. "de" "DE"))
+      fr (NumberFormat/getInstance (Locale. "fr" "FR"))
+      ch (NumberFormat/getInstance (Locale. "de" "CH"))]
+  (defn- parse-plain-number [number-separators s]
+    (case number-separators
```

Comment:
Calling `(get-number-separators)` hits the app DB and is expensive. I moved the call up so that it's only called once per upload, not once per number to be parsed.